src/Features/LanguageServer/Protocol/Handler/PullHandlers/VersionedPullCache`1.cs

```diff
@@ -28,8 +31,9 @@ public VersionedPullCache(string uniqueKey)
             CancellationToken cancellationToken)
         {
             return GetNewResultIdAsync(
-                documentToPreviousDiagnosticParams,
-                document,
+                documentToPreviousDiagnosticParams.ToDictionary(kvp => new ProjectOrDocumentId(kvp.Key.Id), kvp => kvp.Value),
```

Comment:
I previously suggested we unify these caches, however now it is becoming annoying.  Since this PR is large I will un-unify these in a followup